test/remote/gateways/remote_cashnet_test.rb

```diff
@@ -22,6 +22,15 @@ def test_successful_purchase
     assert_equal 'Success', purchase.message
   end
 
+  def test_successful_purchase_with_multiple_items
+    options = @options.merge({itemcode: "FEE", itemcode2: "LOBSTER", itemcode3: "CODES", amount2: 1234, amount3: 4321})
```

Comment:
Instead of allowing for an infinite number of separate `itemcode` fields, I think what we want is to allow for an `item_codes` field that holds an object like:
```
"item_codes": {
  "item_code": "something",
  "amount": 100,
  "item_code_2": "something else",
  "amount_2": 50
}
```

Then, in the gateway adapter, `add_invoice` could look for `options[:item_codes]` and unpack it if it is present, mapping each field from snake_case to what the CASHNet API expects. If there is no `item_codes`, the method could do essentially what it does now with `post[:itemcode] = (options[:item_code] || @options[:default_item_code])`.

(Also, it's a real shame that CASHNet's API calls the first fields `itemcode` and `amount` instead of `itemcode1` and `amount1`)

Reply:
@dsmcclain this seems like a good approach! My only concern is that even in his configuration, how can we NOT allow an infinite number of `itemcode` fields? Should we arbitrarily choose a maximum number of `itemcode` fields that we will map over?

I'm thinking that using your method, the logic would look like this:

```
def add_invoice(post, money, options)
    if options[:itemcodes].present?
        ###iterate through this map of itemcodes and amounts and add them to our post
    else
        post[:itemcode] = (options[:item_code] || @options[:default_item_code])
end
```

How does this solve the "infinite number of separate itemcode fields" problem?

Reply:
(also, I agree with you. not including that `itemcode1` and `amount1` makes iterating over those fields wayyyy clunkier)

Reply:
@peteroas Yes, your pseudocode is precisely what I am imagining. Except the `options` fields would be snake_case like `item_codes`.

> How does this solve the "infinite number of separate itemcode fields" problem?

Good question. Doing it this way provides a few advantages:
* It makes the request more organized. Having a container field like `item_codes` makes it easier for humans to read and write the requests. You are right that we could still add an unlimited number of item codes, but as the number of itemcodes grows the size of the request at the root level only grows by one.
* It makes it easier to code against the requests, because we can sequester the `item_codes` object and perform whatever parsing and logic we want with it without having to handle the entire request body along with it.
* As an example of the above, Spreedly Core defines every field that is passed along to the Active Merchant gateway adapter. If we put each itemcode and amount field at the root level, we would have to enumerate every field or else create some sort of wildcard to handle it. This way, we only have to specify one field: `item_codes`.